packages/blockchain-wallet-v4-frontend/src/modals/Brokerage/Banks/Withdraw/EnterAmount/index.tsx

```diff
@@ -20,96 +20,89 @@ import Failure from './template.failure'
+      props.custodialActions.fetchCustodialBeneficiaries(props.fiatCurrency)
+      props.withdrawActions.fetchWithdrawalLock()
     }
-  }
+  }, [props.fiatCurrency])
 
-  handleSubmit = () => {
-    const { defaultBeneficiary } = this.props.data.getOrElse({} as SuccessStateType)
-    const { defaultMethod } = this.props
-    const beneficiary = defaultBeneficiary || this.props.beneficiary
+  const handleSubmit = () => {
+    const { defaultBeneficiary } = props.data.getOrElse({} as SuccessStateType)
+    const { defaultMethod } = props
+    const beneficiary = defaultBeneficiary || props.beneficiary
 
     if (!beneficiary && !defaultMethod) return
 
     if (defaultMethod) {
-      this.props.withdrawActions.setStep({
-        amount: this.props.formValues.amount,
+      props.withdrawActions.setStep({
+        amount: props.formValues.amount,
         defaultMethod: defaultMethod as BankTransferAccountType,
         step: WithdrawStepEnum.CONFIRM_WITHDRAW
       })
-    } else if (defaultBeneficiary || this.props.beneficiary) {
-      this.props.withdrawActions.setStep({
-        amount: this.props.formValues.amount,
+    } else if (defaultBeneficiary || props.beneficiary) {
+      props.withdrawActions.setStep({
+        amount: props.formValues.amount,
         beneficiary,
         step: WithdrawStepEnum.CONFIRM_WITHDRAW
       })
     }
   }
 
-  handleBankSelection = (
+  const handleBankSelection = (
     userData: UserDataType,
     beneficiary?: BeneficiaryType | BankTransferAccountType
   ) => {
     if (!beneficiary) {
-      this.props.simpleBuyActions.showModal('WithdrawModal')
+      props.simpleBuyActions.showModal('WithdrawModal')
       if (userData.tiers.current === 2) {
-        return this.props.simpleBuyActions.setStep({
+        return props.simpleBuyActions.setStep({
           addBank: true,
           displayBack: false,
-          fiatCurrency: this.props.fiatCurrency,
+          fiatCurrency: props.fiatCurrency,
           step: 'BANK_WIRE_DETAILS'
         })
       }
-      return this.props.simpleBuyActions.setStep({
+      return props.simpleBuyActions.setStep({
         step: 'KYC_REQUIRED'
       })
     }
 
-    this.props.withdrawActions.setStep({
-      fiatCurrency: this.props.fiatCurrency,
+    props.withdrawActions.setStep({
+      fiatCurrency: props.fiatCurrency,
       step: WithdrawStepEnum.BANK_PICKER
     })
   }
 
-  render() {
-    return this.props.data.cata({
-      Failure: () => <Failure {...this.props} />,
-      Loading: () => <Loading />,
-      NotAsked: () => <Loading />,
-      Success: (val) => (
-        <Success
-          {...this.props}
-          {...val}
-          onSubmit={this.handleSubmit}
-          handleBankSelection={this.handleBankSelection}
-        />
-      )
-    })
-  }
+  return props.data.cata({
+    Failure: () => <Failure {...props} />,
+    Loading: () => <Loading />,
+    NotAsked: () => <Loading />,
+    Success: (val) => (
+      <Success
+        {...props}
```

Comment:
We do this all around app, but would be good practice to stop it and prevent spreading all props from top down, better to pass only necessary props.

Reply:
Why?